### DISCUSSION TITLE
How to edit documents?

### DISCUSSION THREAD

**Discussion by JozanLeClerc** [outline/outline#3911] [General] [answered]
Hi, noob question maybe.

I am not sure if this is a bug or not but I'm not able to edit documents on the web front-end. I run a self-hosted Outline installation for my company (docker version) with Keycloak as OIDC.

I can create collections and documents, set the title and that's pretty much it. I was able to fully import a markdown file but I can't edit it either on the webpage. I tried several browsers (Firefox, Chrome, Chrome-based) but it doesn't make a difference.

I am new to Outline so I'm not sure if this is normal behavior or if this is a bug. Am I doing something wrong?

**Comment by tommoor** [comment, marked as answer]
Most likely websockets not working, see various other discussions:
https://github.com/outline/outline/discussions/2222

**Comment by apoorv-mishra** [comment]
Do you see the _Edit_ option at all in document menu? @JozanLeClerc 

![Screenshot from 2022-08-03 13-07-21](https://user-images.githubusercontent.com/15694566/182551546-a1d5eaaa-0bb0-4888-b55a-29fa86444bb1.png)

If not, then try disabling _Collaborative editing_ under _Setting > Features_.

**Reply by JozanLeClerc** [reply]
Hi aproov, thanks for showing me this option. I disabled *Collaborative editing* and this *Edit* option did appear. However edition was already made possible thanks to the steps listed on my answer below.

**Comment by JozanLeClerc** [comment]
Thanks for pointing me to the right direction @tommoor, indeed my nginx proxy was badly configured. Here is my old and new nginx configs for anyone interested.

Old config:

```conf
server {
        listen 80;
        listen [::]:80;
        server_name doc.mydomain;
        location / {
                proxy_pass http://doc.mydomain:3000/;
        }
}
```

New config (edition works):

```conf
server {
        listen 80;
        listen [::]:80;
        server_name doc.mydomain;
        location / {
                proxy_pass http://doc.mydomain:3000/;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header Host $host;
                proxy_redirect off;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";
        }
}
```
